<a href="https://colab.research.google.com/github/LathaKPAIML/AIML/blob/master/Fivefold_Hackathon_Sep_2020.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import pylab as pl
%matplotlib inline
import sklearn

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
info_train = pd.read_csv('info_train.csv')
meta_train = pd.read_csv('metadata_train.csv')
rate_train = pd.read_csv('ratings_train.csv')

## Shape of dataset

In [ ]:
print("Shape of info_train")
print(info_train.shape)
print("Shape of meta_train")
print(meta_train.shape)
print("Shape of rate_train")
print(rate_train.shape)

Shape of info_train
(12755, 10)
Shape of meta_train
(12755, 6)
Shape of rate_train
(12755, 4)


Info_train has 10 features, Meta_train has 6 features, Rate_train has 4 features

## Data Exploration (Primitive Analysis)

In [ ]:
info_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12755 entries, 0 to 12754
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   URL              12755 non-null  object 
 1   Name             12755 non-null  object 
 2   game_id          12755 non-null  object 
 3   Subtitle         3938 non-null   object 
 4   Icon URL         12755 non-null  object 
 5   Description      12755 non-null  object 
 6   Developer        12755 non-null  object 
 7   Genres           12755 non-null  object 
 8   Primary Genre    12755 non-null  object 
 9   gamestore_price  12738 non-null  float64
dtypes: float64(1), object(9)
memory usage: 996.6+ KB


In info_train dataset, we have only 3938 non-null objects under "Subtitle", and also some values are of null type in "gamestore_price"

In [ ]:
meta_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12755 entries, 0 to 12754
Data columns (total 6 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   In-app Purchases              5770 non-null   object 
 1   Languages                     12711 non-null  object 
 2   Size                          12755 non-null  float64
 3   Original Release Date         12755 non-null  object 
 4   Current Version Release Date  12755 non-null  object 
 5   game_id                       12755 non-null  object 
dtypes: float64(1), object(5)
memory usage: 598.0+ KB


The features 'In-app purchases', "languages" have null values in meta_train dataset

In [ ]:
rate_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12755 entries, 0 to 12754
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Average User Rating  5687 non-null   float64
 1   User Rating Count    5687 non-null   float64
 2   Age Rating           12755 non-null  object 
 3   game_id              12755 non-null  object 
dtypes: float64(2), object(2)
memory usage: 398.7+ KB


"Average User Rating" and "User Rating Count" are having many null  NA values in Ratings_train dataset.

## Datasets Merging

Let us merge all the three datasets together so as to perform all our operations on a single dataset.

In [ ]:
dfone = pd.merge(info_train, meta_train, on="game_id")

In [ ]:
df  = pd.merge(dfone, rate_train, on = "game_id")

In [ ]:
df.shape

(12755, 18)

In [ ]:
df.columns

Index(['URL', 'Name', 'game_id', 'Subtitle', 'Icon URL', 'Description',
       'Developer', 'Genres', 'Primary Genre', 'gamestore_price',
       'In-app Purchases', 'Languages', 'Size', 'Original Release Date',
       'Current Version Release Date', 'Average User Rating',
       'User Rating Count', 'Age Rating'],
      dtype='object')

In [ ]:
df.head()

URL  ... Age Rating
0  https://apps.apple.com/us/app/lottery-life-mon...  ...        12+
1  https://apps.apple.com/us/app/cute-little-unic...  ...         4+
2  https://apps.apple.com/us/app/its-tanks/id5688...  ...         9+
3   https://apps.apple.com/us/app/dunes/id1453578024  ...         4+
4  https://apps.apple.com/us/app/cribbage-square-...  ...         4+

[5 rows x 18 columns]

In [ ]:
ID = df["game_id"]

##  Feature Engineering

As the columns "URL", "Icon_URL" have no meaning and does not imply anythong, we shall remove them.

In [ ]:
df = df.drop(['URL','Icon URL','game_id'], axis=1 )

In [ ]:
df.dtypes

Name                             object
Subtitle                         object
Description                      object
Developer                        object
Genres                           object
Primary Genre                    object
gamestore_price                 float64
In-app Purchases                 object
Languages                        object
Size                            float64
Original Release Date            object
Current Version Release Date     object
Average User Rating             float64
User Rating Count               float64
Age Rating                       object
dtype: object

In [ ]:
df.isnull().sum()

Name                               0
Subtitle                        8817
Description                        0
Developer                          0
Genres                             0
Primary Genre                      0
gamestore_price                   17
In-app Purchases                6985
Languages                         44
Size                               0
Original Release Date              0
Current Version Release Date       0
Average User Rating             7068
User Rating Count               7068
Age Rating                         0
dtype: int64